devices/stm32g070.yaml

```diff
@@ -8,6 +8,126 @@ DMA:
     IFCR:
       access: write-only
 
+_delete:
+  - TIM3
+  - LPTIM1
+  - LPTIM2
+  - COMP
+  - DAC
+  - LPUART
+  - UCPD1
+  - UCPD2
+  - SYSCFG_VREFBUF
+  - RNG
+  - HDMI_CEC
+
+# Remove timer 3 (above) and rename TIM2 to TIM3. Because TIM3 was derive from TIM2
+_modify:
+  TIM2:
+    name: TIM3
+    baseAddress: "0x40000400"
+# After changing the name, remove TIM2 interrupt from TIM3 block (it already has the TIM3 interrupt)
+TIM3:
+  _delete:
+    _interrupts:
+      - TIM2
+
+RCC:
+  AHBRSTR:
+    _delete:
+      - AESRST
+      - RNGRST
+  APBRSTR1:
+    _delete:
+      - TIM2RST
+      - LPUART1RST
+      - CECRST
+      - UCPD1RST
+      - UCPD2RST
+      - DAC1RST
+      - LPTIM2RST
+      - LPTIM1RST
+  AHBENR:
+    _delete:
+      - AESEN
+      - RNGEN
+  APBENR1:
+    _delete:
+      - TIM2EN
+      - LPUART1EN
+      - CECEN
+      - UCPD1EN
+      - UCPD2EN
+      - DAC1EN
+      - LPTIM2EN
+      - LPTIM1EN
+  AHBSMENR:
+    _delete:
+      - AESSMEN
+      - RNGSMEN
+  APBSMENR1:
+    _delete:
+      - TIM2SMEN
+      - LPUART1SMEN
+      - CECSMEN
+      - UCPD1SMEN
+      - UCPD2SMEN
+      - DAC1SMEN
+      - LPTIM2SMEN
+      - LPTIM1SMEN
+  CCIPR:
+    _delete:
+      - CECSEL
+      - LPUART1SEL
+      - LPTIM2SEL
+      - RNGSEL
+      - RNGDIV
+      - LPTIM1SEL
+
+EXTI:
+  _delete:
+    - VERR
+    - IPIDR
+    - SIDR
+
+# Fix interrupts, see RM454 Rev3 11.3 Table 42
+
+ADC:
+  _modify:
+    _interrupts:
+      ADC_COMP:
+        name: ADC
+        description: ADC Interrupt
+
+TIM1:
+  _modify:
+    _interrupts:
+      TIM1_BRK_UP_TRG_COMP:
+        name: TIM1_BRK_UP_TRG_COM
+        description: IM1 break, update, trigger and commutation interrupts
+
+# TIM 6 and 7 interrupts
+TIM*:
+  _modify:
+    _interrupts:
+      TIM6_DAC_LPTIM1:
+        name: TIM6
+        description: TIM6 global interrupt
+      TIM7_LPTIM2: #TODO: does not have any effect (in TIM7) see bug below
+        name: TIM7
+        description: TIM7 global interrupt
+
+# TODO: crashes svdtools.patch.MissingPeripheralError: Could not find USART3.
+# maybe bug in svdtools v1.6
+# USART3:
+#   _modify:
+#     _interrupts:
+#       USART3_USART4_LPUART1:
+#         name: USART3_USART4
+#         description: USART3 + USART4 interrupt
```

Comment:
I guess this needs to be addressed before merging.

Reply:
I believe this is related to https://github.com/stm32-rs/svdtools/pull/13 which was reverted. If a peripheral is `derivedFrom` another one, then the patching tool cannot seem to match on the peripheral name. Maybe there is a workaround but I couldn't figure it out yet.

```xml
<peripheral derivedFrom="USART1">
    <name>USART3</name>
    <baseAddress>0x40004800</baseAddress>
    <interrupt>
      <name>USART3_USART4_LPUART1</name>
      <description>USART3 + USART4 + LPUART1</description>
      <value>29</value>
    </interrupt>
  </peripheral>
```

So the result is that the STM32G070 crate will still have `USART3` erroneously available through its PAC. It is not a big deal I think. When svdtools supports `derivedFrom` we can still remove it.

Reply:
@adamgreig do you know if it is possible to do it differently or should we just let it be as it is?

Reply:
Wait this was only about the interrupt _name_. Which should be `USART3_USART4` instead of `USART3_USART4_LPUART1`since there is no `LPUART1` peripheral on this part.

So the issue here is: "How to modify properties of a _`derivedFrom` peripheral_ "

Reply:
Interrupts don't actually need to be associated with the correct peripheral so in this case it might be simplest to just make sure the base peripheral has all relevant interrupts and the derived peripheral has none, which should be doable with current svdtools without needing to modify the derived peripheral.

Reply:
Removing them and re-adding seems to work!